src/gproc_pool.erl

```diff
@@ -570,6 +570,9 @@ do_wait({busy_wait, Ref} = W) ->
     case erlang:read_timer(Ref) of
 	false ->
 	    erlang:cancel_timer(Ref),
+            receive {timeout, Ref, _} -> ok
```

Comment:
Hmm, it would seem as if `read_timer(Ref) -> false` already says that the timer has expired. So canceling it after that should be redundant. Flushing the message could then be done directly.

Reply:
Agreed and force-pushed.

Reply:
@uwiger any chance of taking a further look? Thanks.